### PR TITLE
CleanLearning = Machine Learning with cleaned data

### PR DESCRIPTION
**Pull request by cgnorthcutt**
## Rename `LearningWithNoisyLabels` to `CleanLearning` everywhere.

Users should associate `clean learning` with the notion of doing machine learning (of any kind) with cleaned data. `cleanlab` is the tool to perform `clean learning`.

* This change enables the nomenclature needed for `cleanlab` to generalize beyond just learning with noisy labels, but to learning with cleaned data in general.
* As a side note, users can feel free to use `confident learning` and `clean learning` interchangeably.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,14 +129,14 @@ class YourFavoriteModel(BaseEstimator): # Inherits sklearn base classifier
         pass
 
 # Now you can use your model with `cleanlab`. Here's one example:
-from cleanlab.classification import LearningWithNoisyLabels
-lnl = LearningWithNoisyLabels(clf=YourFavoriteModel())
+from cleanlab.classification import CleanLearning
+lnl = CleanLearning(clf=YourFavoriteModel())
```
Comment (jwmueller) [inline README.md:133]: We should probably change the name of the variable everywhere now. `lnl` doesn't really make sense as a variable name anymore.  I propose we always name the sklearn Estimator to be a variable `model` (sometimes it's currently called `clf` but I'd replace with `model` throughout)  and the CleanLearning-wrapped estimator to be a variable `cl_model` (in all examples, tutorials, readme, etc.)

Comment (cgnorthcutt) [reply README.md:133]: how about replacing `lnl` everywhere with `cl`. I like it since it stands for `cleanlab` too

Comment (cgnorthcutt) [reply README.md]: fixed here [ef92446](https://github.com/cleanlab/cleanlab/pull/177/commits/ef92446022cb4754d537f8737ec3dc77772b8856)

File: README.md [hunk 2]
```diff
@@ -129,14 +129,14 @@ class YourFavoriteModel(BaseEstimator): # Inherits sklearn base classifier
         pass
 
 # Now you can use your model with `cleanlab`. Here's one example:
-from cleanlab.classification import LearningWithNoisyLabels
-lnl = LearningWithNoisyLabels(clf=YourFavoriteModel())
-lnl.fit(train_data, train_labels_with_errors)
+from cleanlab.classification import CleanLearning
+cl = CleanLearning(clf=YourFavoriteModel())
+cl.fit(train_data, train_labels_with_errors)
```
Comment (jwmueller) [inline README.md:134]: Maybe add comment that: all of the methods in `clf` (such as `predict()`) will be available in `cl` as well.

File: cleanlab/classification.py [hunk 3]
```diff
@@ -51,9 +51,9 @@
 
 Examples
 --------
->>> from cleanlab.classification import LearningWithNoisyLabels
+>>> from cleanlab.classification import CleanLearning
 >>> from sklearn.linear_model import LogisticRegression as LogReg
->>> rp = LearningWithNoisyLabels(clf=LogReg()) # Pass in any classifier.
+>>> rp = CleanLearning(clf=LogReg()) # Pass in any classifier.
```
Comment (jwmueller) [inline cleanlab/classification.py:56]: should be `cl` instead of `rp`

**Review by jwmueller — APPROVED** [review]
LGTM, just change variable `rp` -> `cl` as well.

**Comment by jwmueller** [comment]
Also note your commit failed the notebook linter:

docs/source/tutorials/tabular.ipynb [cell 37]: unnecessary trailing newline

File: README.md (hunk 2 above)
Comment (cgnorthcutt) [reply README.md:134]: 👍

File: cleanlab/classification.py (hunk 3 above)
Comment (cgnorthcutt) [reply cleanlab/classification.py:56]: good catch. fixed

**Review by JohnsonKuan — APPROVED** [review]
LGTM too

**Review by anishathalye — APPROVED** [review]
LGTM. Great idea for the new name!
